src/pybind/mgr/nfs/export.py

```diff
@@ -424,63 +398,204 @@ def _create_user_key(self, entity, path, fs_name, fs_ro):
+                    raise NFSException(f'Failed to create user {uid}')
+            j = json.loads(out)
+            # FIXME: make this more tolerate of unexpected output?
+            access_key = j['keys'][0]['access_key']
+            secret_key = j['keys'][0]['secret_key']
+
+            ex_id = self._gen_export_id(cluster_id)
+            if clients:
+                access_type = "none"
+            elif read_only:
                 access_type = "RO"
+            else:
+                access_type = "RW"
             ex_dict = {
-                    'path': self.format_path(path),
-                    'pseudo': pseudo_path,
-                    'cluster_id': cluster_id,
-                    'access_type': access_type,
-                    'fsal': {"name": "CEPH", "user_id": user_id,
-                             "fs_name": fs_name, "sec_label_xattr": ""},
-                    'clients': []
-                    }
+                'path': self.format_path(bucket),
+                'pseudo': pseudo_path,
+                'cluster_id': cluster_id,
+                'access_type': access_type,
+                'squash': squash,
+                'fsal': {
+                    "name": "RGW",
+                    "user_id": uid,
+                    "access_key_id": access_key,
+                    "secret_access_key": secret_key,
+                },
+                'clients': clients
+            }
             export = Export.from_dict(ex_id, ex_dict)
-            export.fsal.cephx_key = key
-            self._save_export(export)
+            export.validate(self.mgr)
+            self._save_export(cluster_id, export)
             result = {
-                    "bind": pseudo_path,
-                    "fs": fs_name,
-                    "path": path,
-                    "cluster": cluster_id,
-                    "mode": access_type,
-                    }
+                "bind": pseudo_path,
+                "path": bucket,
+                "cluster": cluster_id,
+                "mode": access_type,
+                "squash": squash,
+            }
             return (0, json.dumps(result, indent=4), '')
         return 0, "", "Export already exists"
 
-    def update_export(self, new_export):
-        old_export, update_user_caps = self._validate_export(new_export)
-        if update_user_caps:
-            self._update_user_id(new_export['path'], new_export['access_type'],
-                                 new_export['fsal']['fs_name'], new_export['fsal']['user_id'])
-        new_export = Export.from_dict(new_export['export_id'], new_export)
-        new_export.fsal.cephx_key = old_export.fsal.cephx_key
-        self._update_export(new_export)
-        export_ls = self.exports[self.rados_namespace]
-        if old_export not in export_ls:
-            # This happens when export is fetched by ID
-            old_export = self._fetch_export(old_export.pseudo)
-        export_ls.remove(old_export)
+    def update_export_1(
```

Comment:
Please rename this to 
```suggestion
    def update_export(
```

Reply:
And other update export method to `_save_modified_export` or something else
https://github.com/ceph/ceph/blob/84b025b7584c2627e60ef20af7c4aac3be8dc87a/src/pybind/mgr/nfs/export.py#L262-L266